extensions/integration-tests/testfmks/src/stress.ts

```diff
@@ -0,0 +1,289 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+/**
+ * This module contains all the definitions for Stress decorators and the utility functions and definitions thereof
+*/
+import { Min, Max, IsInt, validate } from 'class-validator';
+import 'mocha';
+import { AssertionError } from 'assert';
+import { getSuiteType, SuiteType } from './utils';
+import assert = require('assert');
+import { isString } from 'util';
+
+const logPrefix = 'testfmks:stress';
+const debug = require('debug')(logPrefix);
+const trace = require('debug')(`${logPrefix}:trace`);
+/**
+ * Subclass of Error to wrap any Error objects caught during Stress Execution.
+ */
+export class StressError extends Error {
+	inner: Error | any;
+	static code: string = 'ERR_STRESS';
+
+	constructor(error?: any) {
+		super();
+		this.name = StressError.code;
+		this.inner = error;
+		if (error instanceof Error) {
+			this.message = error.message;
+			this.stack = error.stack;
+		} else if (error instanceof String) {
+			this.message = error.valueOf();
+			try {
+				throw new Error();
+			} catch (e) {
+				this.stack = e.stack;
+			}
+		} else if (isString(error)) {
+			this.message = error;
+			try {
+				throw new Error();
+			} catch (e) {
+				this.stack = e.stack;
+			}
+		} else {
+			this.message = 'unknown stress error';
+		}
+	}
+}
+
+/**
+ * Defines an interface to specify the stress options for stress tests.
+ * @param runtime - the number of minutes for which the stress runs. Once this 'runtime' expires stress is terminated even if we have not exceeded {@link iterations} count yet. NYI. This is here for future use to allow really long running stress tests. Default value is provided by environment variable: StressRuntime and if undefined then by {@link DefaultStressOptions}.
```

Comment:
what is NYI?

Reply:
The runtime is in minutes, the stress class has a runtime which is in seconds, and sleep is in ms. Can we make that more consistent?

Reply:
sleep will stay in milliseconds. That is consistent with most timing apis in the language and most appropriate unit for. The descrepency between stress class and here is a doc bug in one place which I will fix. It will be seconds everywhere.

Reply:
NYI=Not Yet Implemented.

Reply:
removed minutes and made it seconds everywhere for runtime parameter